### PR TITLE
feat(ButtonV2): new component

### PR DESCRIPTION
**Pull request by fabienhebert**
## Type

- Feature

### Summarise concisely:

New Button component (ButtonV2)

### CODE REVIEW

**Review by PetmePant — APPROVED** [review]

File: packages/ui/src/components/ButtonV2/__stories__/ShowcaseTemp.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import type { ComponentStory } from '@storybook/react'
+import { Fragment } from 'react'
+import { ButtonV2, buttonVariants } from '..'
+import { Stack, Table, Text } from '../..'
+import { SENTIMENTS } from '../../../theme'
+
+export const ShowcaseTemp: ComponentStory<typeof ButtonV2> = () => {
```
Comment (matthprost) [inline packages/ui/src/components/ButtonV2/__stories__/ShowcaseTemp.stories.tsx:7]: Reminder to remove this story

File: packages/ui/src/components/ButtonV2/__stories__/Showcase.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import type { ComponentStory } from '@storybook/react'
+import { ButtonV2, buttonVariants } from '..'
+import { Stack, Table, Text } from '../..'
+import { SENTIMENTS } from '../../../theme'
+
+export const Showcase: ComponentStory<typeof ButtonV2> = () => {
```
Comment (matthprost) [inline packages/ui/src/components/ButtonV2/__stories__/Showcase.stories.tsx:6]: Reminder to remove this story

File: packages/ui/src/components/Button/__stories__/index.stories.tsx [hunk 3]
```diff
@@ -12,6 +12,9 @@ export default {
   ],
   parameters: {
     docs: {
+      deprecated: true,
+      deprecatedReason:
```
Comment (matthprost) [inline packages/ui/src/components/Button/__stories__/index.stories.tsx:16]: You also need to add `@deprecated` in Button `index.tsx`

File: packages/ui/src/components/ButtonV2/index.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+  padding: 0 ${({ theme, size }) => theme.space[SIZE_SPACING_KEY[size]]};
+  flex-direction: ${({ iconPosition }) =>
+    iconPosition === 'right' ? 'row-reverse' : 'row'};
+  gap: ${({ theme }) => theme.space['1']};
+  border-radius: ${({ theme }) => theme.radii.default};
+  align-items: center;
+  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
+  width: ${({ fullWidth }) => (fullWidth ? '100%' : 'auto')};
+  justify-content: center;
+  outline-offset: 2px;
```
Comment (matthprost) [inline packages/ui/src/components/ButtonV2/index.tsx:61]: I think we can use space here from theme

```suggestion
  outline-offset: ${({ theme }) => theme.space['0.25']};
```

**Review by matthprost — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/Button/__stories__/index.stories.tsx [hunk 5]
```diff
@@ -12,6 +12,9 @@ export default {
...
+        'This component is deprecated please use ButtonV2 instead.',
       description: {
```
Comment (matthprost) [inline packages/ui/src/components/Button/__stories__/index.stories.tsx:18]: You can add the migration link as you did a migration page:

```suggestion
  parameters: {
    docs: {
      deprecated: true,
      deprecatedReason:
        'This component is deprecated please use ButtonV2 instead.',
      migrationLink: '?path=/docs/migrations-button-v1-to-buttonv2--page'
      description: {
```

**Review by matthprost — CHANGES_REQUESTED** [review]

**Review by matthprost — APPROVED** [review]

File: packages/ui/src/components/ButtonV2/index.tsx [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+  >
+>
+const StyledButton = styled('button', {
+  shouldForwardProp: prop =>
+    !['size', 'sentiment', 'iconPosition', 'fullWidth'].includes(prop),
+})<StyledButtonProps>`
+  display: inline-flex;
+  height: ${({ size }) => `${SIZE_HEIGHT[size]}px`};
+  padding: 0 ${({ theme, size }) => theme.space[SIZE_SPACING_KEY[size]]};
+  flex-direction: ${({ iconPosition }) =>
```
Comment (ModuloM) [inline packages/ui/src/components/ButtonV2/index.tsx:53]: As children is `ReactNode` type, you could eventually pass a fragment with two elements in it. Will they be reverse too?
Ex. 
```javascript
<Button iconPosition="right" icon="pencil">
  <>
    <Text variant="body" prominence="stronger" color="danger">Carreful!</Text>
    <Text variant="body" prominence="strong" color="neutral">Reversed text</Text>
  </>
</Button>
```

Comment (fabienhebert) [reply packages/ui/src/components/ButtonV2/index.tsx:53]: Nice catch, Now children is wrapped in a div if children isn't a simple string

**Pull request merged**